variables/templates_test.go

```diff
@@ -0,0 +1,168 @@
+This is a test file for templating.
+
+  ###PREFIX_VAR_REPLACE_ME###
+  ###PREFIX_CONST_REPLACE_ME###
+  ###PREFIX_APP_REPLACE_ME###
+  ###PREFIX_NON_EXIST###
+`
+var simple = `
+This is a test file for templating.
+
+  VAR_REPLACED
+  CONST_REPLACED
+  APP_REPLACED
+  ###PREFIX_NON_EXIST###
+`
+var multiline = `
+This is a test file for templating.
+
+  VAR_REPLACED
+VAR_SECOND
+  CONST_REPLACED
+CONST_SECOND
+  APP_REPLACED
+APP_SECOND
+  ###PREFIX_NON_EXIST###
+`
+var autoIndent = `
+This is a test file for templating.
+
+  VAR_REPLACED
+  VAR_SECOND
+  CONST_REPLACED
+  CONST_SECOND
+  APP_REPLACED
+  APP_SECOND
+  ###PREFIX_NON_EXIST###
+`
+var file = `
+This is a test file for templating.
+
+  The contents of this file become the template value
+  CONSTs Don't Support File
+  The contents of this file become the template value
+  ###PREFIX_NON_EXIST###
+`
+
+func TestReplaceTextTemplate(t *testing.T) {
+	type test struct {
+		vc           VariableConfig
+		path         string
+		wantErr      error
+		wantContents string
+	}
+
+	tests := []test{
+		{
+			vc:           VariableConfig{setVariableMap: SetVariableMap{}, applicationTemplates: map[string]*TextTemplate{}},
+			path:         "non-existent.test",
+			wantErr:      errors.New("open non-existent.test: no such file or directory"),
+			wantContents: start,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "VAR_REPLACED"},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONST_REPLACED"}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "APP_REPLACED"},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: simple,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "VAR_REPLACED\nVAR_SECOND"},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONST_REPLACED\nCONST_SECOND"}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "APP_REPLACED\nAPP_SECOND"},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: multiline,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "VAR_REPLACED\nVAR_SECOND", Variable: Variable{AutoIndent: true}},
+				},
+				constants: []Constant{{Name: "REPLACE_ME", Value: "CONST_REPLACED\nCONST_SECOND", AutoIndent: true}},
+				applicationTemplates: map[string]*TextTemplate{
+					"###PREFIX_APP_REPLACE_ME###": {Value: "APP_REPLACED\nAPP_SECOND", AutoIndent: true},
+				},
+			},
+			path:         testPath,
+			wantErr:      nil,
+			wantContents: autoIndent,
+		},
+		{
+			vc: VariableConfig{
+				templatePrefix: "PREFIX",
+				setVariableMap: SetVariableMap{
+					"REPLACE_ME": {Value: "file.test", Variable: Variable{Type: FileVariableType}},
```

Comment:
We should cleanup file.test automatically, also should be removed from the PR

Reply:
`file.test` never changes and felt it was better to have it in a file than in the test (since wrapper code would be needed to make / delete it.  I _can_ remove it though if you really want me to

Reply:
(also I intentionally committed it :sweat_smile: `git add variables/file.test -f` lol

Reply:
Ah I see it's an actual file variable. Initially I just assumed you accidentally added it since it was in the git ignore. Yeah I'm cool with keeping it

Reply:
I think we can help avoid confusion by adding the file into a directory called testdata
> Directory and file names that begin with "." or "_" are ignored by the go tool, as are directories named "testdata". 
https://pkg.go.dev/cmd/go#hdr-Package_lists_and_patterns

Reply:
perfect will do that